File: 프로그래머스/1/155652. 둘만의 암호/둘만의 암호.js


function solution(s, skip, index) {
    
    const aToz = 'abcdefghijklmnopqrstuvwxyz';
    
    const regex = new RegExp(skip.split('').join('|'), 'g');
    
    const glance = aToz.replace(regex, '');
    
    const converted = [...s].map(str => {
        const pos = glance.indexOf(str);
        
        const moved = pos + index > glance.length - 1 ? (pos + index) % glance.length : pos + index ;
        
        return glance[moved]
    }).join('');
    
    return converted;
}

File: 프로그래머스/1/155652. 둘만의 암호/둘만의 암호.py
def solution(s, skip, index):
    alphabet = "abcdefghijklmnopqrstuvwxyz"
    
    result = []
    
    for char in s:
        # 알파벳에서의 위치 
        curIdx = alphabet.index(char)
        
        # 이동횟수
        count = 0
        
        while(count < index):
            
            curIdx += 1
            
            if alphabet[curIdx % 26] not in skip:
                count += 1
                
        result.append(alphabet[curIdx % 26])
        
    return "".join(result)
    